### PR TITLE
refactor!: changed the hub sequencer and rollapp api to be more useful and readable

### PR DESCRIPTION
**Pull request by ItzhakBokris**
https://www.notion.so/dymension/RollApp-Module-Modifications-e405dde9a7414212948ffb0bfdd32aed

### CODE REVIEW

File: proto/dymension/rollapp/rollapp.proto [hunk 1]
```diff
@@ -8,26 +8,45 @@ import "gogoproto/gogo.proto";
...
+message RollappSummary {
+  // The unique identifier of the rollapp chain.
+  // The rollappId follows the same standard as cosmos chain_id.
+  string rollappId = 1;
+  // Contains details about the latest UpdateState and the latest finalized UpdateState of the Rollapp.
+  LatestStatesSummary latestStatesSummary = 2;
+}
+
+// LatestStateIndexes contains details about the latest UpdateState and the latest finalized UpdateState of a Rollapp.
+message LatestStatesSummary {
```
Comment (mtsitrin) [inline proto/dymension/rollapp/rollapp.proto:46]: should be `LatestStateSummary` and not `LatestStatesSummary`
It's the summary of a **single** state

File: x/rollapp/client/cli/query_rollapp.go [hunk 2]
```diff
@@ -44,8 +44,8 @@ func CmdListRollapp() *cobra.Command {
 
 func CmdShowRollapp() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "show-rollapp [rollapp-id]",
-		Short: "shows a rollapp",
+		Use:   "get [rollapp-id]",
```
Comment (mtsitrin) [inline x/rollapp/client/cli/query_rollapp.go:47]: I think `show` is more standard way

File: proto/dymension/rollapp/query.proto [hunk 3]
```diff
@@ -93,78 +63,29 @@ message QueryAllRollappRequest {
...
 	  uint64 index = 2;
+		uint64 height = 3;
+		bool finalized = 4;
 }
 
 message QueryGetStateInfoResponse {
 	StateInfo stateInfo = 1 [(gogoproto.nullable) = false];
 }
 
 message QueryAllStateInfoRequest {
```
Comment (mtsitrin) [inline proto/dymension/rollapp/query.proto:81]: what it does? bring all the commits of the rollapp??
thousands of commit?
why it's needed?

File: x/rollapp/client/cli/query_state_info.go [hunk 4]
```diff
@@ -51,64 +53,50 @@ func CmdListStateInfo() *cobra.Command {
 
 func CmdShowStateInfo() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   fmt.Sprintf("show-rollapp-state-info [%s] [%s|%s]", FlagRollappId, FlagStateIndex, FlagRollappHeight),
-		Short: "shows a state_info by index or by height",
+		Use:   "state [rollapp-id]",
+		Short: "Query the state associated with the specified rollapp-id and any other flags. If no flags are provided, return the latest state.",
```
Comment (mtsitrin) [inline x/rollapp/client/cli/query_state_info.go:57]: add description for the `finalized` flag

File: x/rollapp/client/cli/query_state_info.go [hunk 5]
```diff
@@ -51,64 +53,50 @@ func CmdListStateInfo() *cobra.Command {
...
 			}
-			argHeight, err := flagSet.GetUint64(FlagRollappHeight)
+			argFinalized, err := flagSet.GetBool(FlagFinalized)
 			if err != nil {
 				return err
 			}
 
-			queryClient := types.NewQueryClient(clientCtx)
+			if (argHeight != 0 && argIndex != 0) || (argHeight != 0 && argFinalized) || (argIndex != 0 && argFinalized) {
+				return fmt.Errorf("only one flag can be use for %s, %s or %s", FlagStateIndex, FlagRollappHeight, FlagFinalized)
```
Comment (mtsitrin) [inline x/rollapp/client/cli/query_state_info.go:79]: add this case to the Tests

File: x/rollapp/client/cli/query_state_info.go (hunk 4 above)
Comment (ItzhakBokris) [reply x/rollapp/client/cli/query_state_info.go:57]: There is a description:
`cmd.Flags().Bool(FlagFinalized, false, "Indicates whether to return the latest finalized state")
`

File: proto/dymension/rollapp/rollapp.proto (hunk 1 above)
Comment (ItzhakBokris) [reply proto/dymension/rollapp/rollapp.proto:46]: This object defined two types of states: latestState and latestFinalizedState

**Pull request merged**